File: LeetCode/90.subsets-ii.cpp
/*
 * @lc app=leetcode id=90 lang=cpp
 *
 * [90] Subsets II
 */

// @lc code=start
class Solution {
public:
   
    vector<vector<int>> ans;
    vector<int> cur;
    set<vector<int>> s;
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {

        sort(nums.begin(), nums.end());

        rec(nums, 0, nums.size());

        return ans;

    }

    void rec(vector<int>& nums, int i, int n)
    {
        if(i==n)
        {
            if(s.find(cur)==s.end())
            {
                ans.push_back(cur);
                s.insert(cur);
            }
            return;
        }

        cur.push_back(nums[i]);
        rec(nums, i+1, n);
        cur.pop_back();
        rec(nums, i+1, n);
    }





  
};
// @lc code=end

File: LeetCode/90.subsets-ii.java
/*
 * @lc app=leetcode id=90 lang=java
 *
 * [90] Subsets II
 */

// @lc code=start
class Solution {


    List<List<Integer>> ans = new ArrayList<>();

    List<Integer> cur = new ArrayList<>();

    Set<List<Integer>> s= new HashSet<>();







    public List<List<Integer>> subsetsWithDup(int[] nums) {

        Arrays.sort(nums);

        rec(nums, 0, nums.length);

        return ans;



       
        
        
    }


    private void rec(int[] nums, int i, int n)
    {
        if(i==n)
        {
            if(!s.contains(cur))
            {
                ans.add(new ArrayList<>(cur));
                s.add()
            }

        }
    }





}
// @lc code=end

